### PR TITLE
DEPS: Bump to 1.26 proper for Python 3.12

### PR DESCRIPTION
**Pull request by lithomas1**
- [ ] closes #xxxx (Replace xxxx with the GitHub issue number)
- [ ] [Tests added and passed](https://pandas.pydata.org/pandas-docs/dev/development/contributing_codebase.html#writing-tests) if fixing a bug or adding a new feature
- [ ] All [code checks passed](https://pandas.pydata.org/pandas-docs/dev/development/contributing_codebase.html#pre-commit).
- [ ] Added [type annotations](https://pandas.pydata.org/pandas-docs/dev/development/contributing_codebase.html#type-hints) to new arguments/methods/functions.
- [ ] Added an entry in the latest `doc/source/whatsnew/vX.X.X.rst` file if fixing a bug or adding a new feature.

### CODE REVIEW

File: .github/workflows/unit-tests.yml [hunk 1]
```diff
@@ -234,7 +234,8 @@ jobs:
           /opt/python/cp311-cp311/bin/python -m venv ~/virtualenvs/pandas-dev
           . ~/virtualenvs/pandas-dev/bin/activate
           python -m pip install --no-cache-dir -U pip wheel setuptools meson[ninja]==1.2.1 meson-python==0.13.1
-          python -m pip install --no-cache-dir versioneer[toml] cython numpy python-dateutil pytz pytest>=7.3.2 pytest-xdist>=2.2.0 pytest-asyncio>=0.17 hypothesis>=6.46.1
+          # No longer possible to easily compile numpy > 1.26 from source
```
Comment (WillAyd) [inline .github/workflows/unit-tests.yml:237]: What makes the source install difficult?

Comment (mroeschke) [reply .github/workflows/unit-tests.yml:237]: Curious about this too. Are the tests not even able to run due to installing numpy?

Comment (lithomas1) [reply .github/workflows/unit-tests.yml:237]: numpy needs a BLAS which is not present on Github Actions to build, by default.

I didn't feel like spending time to fiddle around with the config settings for 32-bit (given its probably going to be removed sometime soon-ish), so I just left it. IMO, it's not the end of the world that its pinned, but if you want me to try to fiddle around, I can set some time towards the end of this week.

Comment (mroeschke) [reply .github/workflows/unit-tests.yml:237]: Are you referring to this failure? https://github.com/pandas-dev/pandas/commit/ebca6dfb8f159ddc00081a463a6d9eda0fc96d8c

Comment (lithomas1) [reply .github/workflows/unit-tests.yml:237]: Ah, nice.

Thanks for the fix!

**Review by mroeschke — APPROVED** [review]

**Pull request merged**